stm32-modules/heater-shaker/simulator/socket_sim_driver.cpp

```diff
@@ -16,10 +16,14 @@ const std::string SOCKET_DRIVER_NAME = "Socket";
 std::unique_ptr<boost::asio::ip::tcp::socket> connect_to_socket(
     std::string host, int port) {
     boost::asio::io_service io_context;
+    boost::asio::ip::tcp::resolver resolver(io_context);
+    auto endpoints = resolver.resolve(host, std::to_string(port));
+    std::string parsed_host =
```

Comment:
I think you should check `endpoints` and print a nice error if it's empty.

Reply:
Is https://github.com/Opentrons/opentrons-modules/pull/224/commits/903c0d426b9bb5f0168f5c1dd335fd0fbbeeabf5 better